astropy/modeling/bounding_box.py

```diff
@@ -17,15 +17,11 @@
 from astropy.utils import isiterable
 
 if TYPE_CHECKING:
-    import sys
     from typing import Any, Callable
 
-    from astropy.units import UnitBase
+    from typing_extensions import Self
```

Comment:
Just to be clear what `ruff` would do: it would just remove this import and add `Self` above once we are at `python>=3.11`, right?

Reply:
See this example:
```
$ cat example.py 
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    import sys
    from typing import Any, Callable

    from typing_extensions import Self
$ ruff check --select=I001,UP035 --fix --target-version=py311 --diff example.py  
--- example.py
+++ example.py
@@ -2,6 +2,5 @@
 
 if TYPE_CHECKING:
     import sys
-    from typing import Any, Callable
-
-    from typing_extensions import Self
+    from collections.abc import Callable
+    from typing import Any, Self

Would fix 3 errors.
```

Reply:
That's really nice!